language/diem-vm/src/adapter_common.rs

```diff
@@ -0,0 +1,311 @@
+use crate::{counters::*, create_access_path, data_cache::StateViewCache};
+use anyhow::Result;
+use diem_state_view::StateView;
+use diem_types::{
+    account_address::AccountAddress,
+    account_config::{self, CurrencyInfoResource, RoleId},
+    transaction::{
+        GovernanceRole, SignatureCheckedTransaction, SignedTransaction, VMValidatorResult,
+    },
+    vm_status::{StatusCode, VMStatus},
+};
+use move_core_types::{
+    identifier::{IdentStr, Identifier},
+    move_resource::MoveStructType,
+    resolver::MoveResolver,
+};
+use move_vm_runtime::session::Session;
+
+use crate::logging::AdapterLogSchema;
+use diem_types::transaction::{Transaction, TransactionOutput, TransactionStatus, WriteSetPayload, TransactionPayload, TransactionArgument};
+use diem_types::access_path::AccessPath;
+use diem_logger::prelude::*;
```

Comment:
I don't think the generic adapter should depend on diem_logger. Also just to mention that @bmwill is planning to get rid of diem_logger altogether and replace it with the `tracing` crate. You may want to use that instead.